opennms-services/src/main/java/org/opennms/netmgt/poller/monitors/BSFMonitor.java

```diff
@@ -232,11 +257,30 @@ public PollStatus poll(MonitoredService svc, Map<String,Object> map) {
             pollStatus = PollStatus.unavailable(e.getMessage());
             LOG.warn("BSFMonitor poll for service '{}' failed with unexpected throwable: {}", svc.getSvcName(), e.getMessage(), e);
         } finally { 
-            bsfManager.terminate();
+            try{
```

Comment:
If you want to be totally correct here, you should loop and put the try/catch inside the loop so that if any undeclareBean() call fails, it won't prevent attempts to undeclare subsequent beans.